ethereum_history_api/circuits/lib/src/rlp/types_test.nr

```diff
@@ -1,5 +1,44 @@
 mod rlp_fragment {
-    mod assert_eq_bytes {
+    mod assert_eq_bytes_length {
+        use crate::rlp::{decode::{STRING, LIST}, types::RlpFragment};
+
+        global bytes = [0x12, 0x34];
+        global fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
+
+        #[test]
+        fn success() {
+            fragment.assert_eq_bytes_length("Field", bytes, bytes, 2)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP type")]
+        fn invalid_type() {
+            let empty = [];
+            let invalid_type_fragment = RlpFragment { offset: 0, length: 0, data_type: LIST };
+
+            invalid_type_fragment.assert_eq_bytes_length("Field", empty, empty, 0)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            let invalid_length_fragment = RlpFragment { offset: 0, length: 3, data_type: STRING };
+
+            invalid_length_fragment.assert_eq_bytes_length("Field", bytes, bytes, 2)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length_given() {
+            fragment.assert_eq_bytes_length("Field", bytes, bytes, 3)
+        }
```

Comment:
This test seems to be an overkill
I don't see - what it tests that the test above does not

Reply:
It tests if we correctly use this argument value. It's the only test where we give different length value than the length of byte array.